### ISSUE TITLE
remote server URI not removable (Error: Invalid remote URI)

### ISSUE THREAD

**Issue by wegmatho** [microsoft/vscode-jupyter#13846]
### Applies To

- [X] Notebooks (.ipynb files)
- [ ] Interactive Window and\/or Cell Scripts (.py files with \#%% markers)

### What happened?

When opening any notebook it immediately tries to connect to an old URI which should not be in the list ("https://DOMAIN.COM/**jupyter2**?token=THETOKEN"). However it is still a valid URI and token (can be used normally via Browser).
It is not possible to remove this setting/URI. Additionally, it does not show in the list ("Select Kernel" menu), only in the logs.

Things I've tried:
- menu "Clear Jupyter Remote Server List"
- completely uninstalling and re-installing the Jupyter Extension

Also, while this happens I cannot use any other remote kernels. In general there seems to be issues with the extension when using multiple remotes and/or standard https port 443 and/or non-standard jupyter base_url (/jupyter, /myjupyter, ...). There are other GitHub issues for this though, I think. I am simply trying to remove the URI from jupyter kernel config/list, which does not seem to be possible.

### VS Code Version

1.79.2

### Jupyter Extension Version

2023.5.1101742258

### Jupyter logs

```shell
10:47:37.502 [info] Start refreshing Interpreter Kernel Picker (1688546857502)
10:47:37.661 [info] Using Pylance
10:47:37.842 [error] Failed to parse remote URI https://DOMAIN.COM/jupyter2?token=THETOKEN [Error: Invalid remote URI
	at qr (c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1101742258-win32-x64\out\extension.node.js:2:1694598)
	at c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1101742258-win32-x64\out\extension.node.js:17:126648
	at async Promise.all (index 1)
	at async c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1101742258-win32-x64\out\extension.node.js:17:126492]

#############
NOTE: following logs, when trying to add another kernel -> notice still tried to use "/jupyter2" again later)
#############

10:58:17.760 [error] Failed to parse remote URI https://DOMAIN.com/jupyter?token=TOKEN [Error: Invalid remote URI
	at qr (c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1101742258-win32-x64\out\extension.node.js:2:1694598)
	at hk.parseUri (c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1101742258-win32-x64\out\extension.node.js:24:530090)
	at c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1101742258-win32-x64\out\extension.node.js:24:528878
	at b.invoke (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at d.deliver (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2121)
	at n.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1729)
	at f._fireDidAccept (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:101:114142)
	at E.$onDidAccept (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:101:112352)
	at s.N (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:11717)
	at s.M (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:11435)
	at s.H (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:10516)
	at s.G (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:9494)
	at c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:111:8282
	at b.invoke (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at d.deliver (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2121)
	at n.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1729)
	at p.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:73:14783)
	at c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:127:32461
	at b.invoke (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:145)
	at d.deliver (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:2121)
	at n.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:63:1729)
	at p.fire (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:73:14783)
	at MessagePortMain.<anonymous> (c:\Users\<username>\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:127:30587)
	at MessagePortMain.emit (node:events:513:28)
	at MessagePortMain._internalPort.emit (node:electron/js2c/utility_init:2:367)]
10:58:19.158 [error] Failed to setup finder for Select Kernel [Error: 'Failed to parse remote URI https://DOMAIN.com/jupyter2?token=TOKEN
	at qr (c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1101742258-win32-x64\out\extension.node.js:2:1694682)
	at c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1101742258-win32-x64\out\extension.node.js:17:126648
	at async Promise.all (index 1)
	at async c:\Users\<username>\.vscode\extensions\ms-toolsai.jupyter-2023.5.1101742258-win32-x64\out\extension.node.js:17:126492]
```

### Running Jupyter locally or remotely?

Remotely

**Comment by DonJayamanne** [comment]
Thank you for filing this issue, I'm sorry you are running into this
We will be releasing a new version of the extension within a day or so and you can try that and let me know if it works.
You could optionally try to isntall the latest pre-release version in VS code insiders,. however based on your comments thats not going to work, as the issue lies with some data thats stored in vs code stable.

Please do let  me know if this issue exists after the next release. (will ping you)

**Comment by wegmatho** [comment]
> Thank you for filing this issue, I'm sorry you are running into this We will be releasing a new version of the extension within a day or so and you can try that and let me know if it works. You could optionally try to isntall the latest pre-release version in VS code insiders,. however based on your comments thats not going to work, as the issue lies with some data thats stored in vs code stable.
> 
> Please do let me know if this issue exists after the next release. (will ping you)

confirmed fixed with VSCode 1.80.0 / Jupyter Extensions v2023.6.1001861915
thanks a lot!

**Comment by DonJayamanne** [comment]
Thanks for getting back to us with a confirmation of the fix.
